src/snkit/network.py

```diff
@@ -603,3 +604,43 @@ def set_precision(geom, precision):
     geom_mapping = mapping(geom)
     geom_mapping['coordinates'] = np.round(np.array(geom_mapping['coordinates']), precision)
     return shape(geom_mapping)
+
+
+def to_networkx(network):
+    """Return a networkx graph
+    """
+    # init graph
+    G = nx.Graph()
+    # get nodes from network data
+    G.add_nodes_from(network.nodes.id.to_list())
+    # get edges from network data
+    edges_as_list = [(network.edges.loc[i].from_id, 
+                      network.edges.loc[i].to_id) 
+                      for i in network.edges.index]
+    # add edges to graph
+    G.add_weighted_edges_from(edges_as_list)
+    return G
+
+
+def connected_components(network,add_id=True,id_col='graph_id'):
```

Comment:
Let's discuss the interface here - it might be clearer to have two functions, one which adds a `component_id` column, another which returns a list/generator of component networkx graphs.

Reply:
I've updated the two above comments as:

https://github.com/amanmajid/snkit/blob/6c506eea7dc6e2a54f5ef097d5d8b87c524f45a5/src/snkit/network.py#L630-L656

Better?

Reply:
Yes, this separation looks good to me